test/fake-api-client.js

```diff
@@ -0,0 +1,50 @@
+export class FakeResource {
+    constructor(resourceType, attributes) {
+        this.resourceType = resourceType;
+        this.attributes = attributes;
+    }
+
+    save() {
+        let saved = this.resourceType.save(this.attributes);
+
+        if (saved) {
+            return Promise.resolve(this);
+        }
+        else {
+            return new Promise((resolve, reject) => { reject(); });
```

Comment:
`reject( new Error('Save failed!') )` would be better here. The tests currently log `Failed to save a queued classification: undefined`.